Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -1048,20 +1052,26 @@ WarpXParticleContainer::PushX (int lev, Real dt)
             Real* AMREX_RESTRICT ux = attribs[PIdx::ux].dataPtr();
             Real* AMREX_RESTRICT uy = attribs[PIdx::uy].dataPtr();
             Real* AMREX_RESTRICT uz = attribs[PIdx::uz].dataPtr();
-            // Loop over the particles and update the position
-            const long np = pti.numParticles();
-            const Real inv_c2 = 1./(PhysConst::c*PhysConst::c);
-            amrex::ParallelFor( np,
+#ifdef WARPX_RZ
+            Real* AMREX_RESTRICT theta = attribs[PIdx::theta].dataPtr();
+#endif
+            // Loop over the particles and update their position
+            amrex::ParallelFor( pti.numParticles(),
```

Comment:
For `PushX`, it seems that the kernels used are known at compile time, so you can write
```
#include <GetAndSetPosition.H>
amrex::ParallelFor(
  // inline GetPosition
  // [...]
);
```
For the momentum push, we'll have several choices (Boris and Vay pushers for the moment), that we wouldn't like to chose at compile time. In this case, would you suggest we do something like that?
```
#include <BorisMomentumPush.H>
#include <VayMomentumPush.H>
if (Boris){
  amrex::ParallelFor(
     // inline BorisPusher
  );
}
if (Vay){
  amrex::ParallelFor(
     // inline VayPusher
  );
}
```
That would create some duplication, but inlining all functions we can use may be a waste.

Reply:
I like Maxence's suggestion.

Reply:
OK, I agree. 
In fact, I am working on the momentum push in C++ right now. 
I can implement @MaxThevenet's suggestion, and then create a separate PR for that.